components/TARGET_PSA/spm/COMPONENT_SPE/spm_main.c

```diff
@@ -18,12 +18,14 @@
 #if defined(COMPONENT_SPM_MAILBOX)
 
 #include "ipc_queue.h"
+#include "sleep_api.h"
 extern ipc_consumer_queue_t *cons_queue;
 // this is the dispatcher thread for dual-chip systems
 int main(void)
 {
     while (true) {
         ipc_queue_drain(cons_queue);
+        hal_deepsleep();
```

Comment:
ipc_queue_drain waits on semaphore with a timeout in case queue is empty. Why are we putting it to sleep manually?

Reply:
It seems that wfi/e aren't the same as deep sleep
For the psoc 6 to go deep sleep(and pass the hal test) all the cores must go deep sleep

Reply:
It should not be the case for asymmetric multiprocessing.
I think we have some misconfiguration in a target port.

Reply:
Whatever the problem is, that isn't the solution.

When RTX is idle, deep sleep should occur as long as no-one is holding the deep sleep lock, and MBED_TICKLESS and DEVICE_LPTICKER are enabled. Check those factors first.

No normal code should ever be calling `hal_sleep` or `hal_deepsleep` manually - it interferes with the RTOS.